lib/kernel/SPIR/generate_spir_wrapper.py

```diff
@@ -768,11 +768,25 @@ def generate_function(name, ret_type, ret_type_ext, multiAS, *args):
 			#   Dv2_cPULocalDv2_c -> Dv2_cPULocalS_
 			pure_arg = pure_arg_type(cast)
 			actual_arg = cast
-			if pure_arg == last_pure_arg and last_pure_arg.startswith("D") and (not last_pure_arg.startswith("Dh")):
-				actual_arg = replace_arg_type(actual_arg, "S_")
-			if pure_arg == last_pure_arg and last_pure_arg == '12memory_order':
-				actual_arg = replace_arg_type(actual_arg, "S4_")
-			last_pure_arg = pure_arg
+
+			replaced = False
+			if len(pure_arg) > 1 and pure_arg != "Dh":
+				for idx, prev_arg in enumerate(saved_pure_args):
```

Comment:
The indentation looks wrong here. Is it because of tabs?

Reply:
yes, this file uses tabs